lib/global-styles.php

```diff
@@ -130,6 +130,14 @@ function_exists( 'gutenberg_is_edit_site_page' ) &&
 			'__experimentalNoWrapper' => true,
 		);
 
+		// Reset existing global styles.
+		foreach ( $settings['styles'] as $key => $style ) {
```

Comment:
@nosolosw Should `$settings['styles']` always be defined before this line? This throws an error in the navigation editor which doesn't have a `style` setting beforehand, but still calls the `gutenberg_experimental_global_styles_settings` function. 

Not entirely sure what the right way to solve that is, so any advice would be appreciated.

Reply:
Dan, I've prepared https://github.com/WordPress/gutenberg/pull/33127

But now I'm wondering if I understood the issue correctly: did you mean that the global styles shouldn't be present in the navigation editor either?

Reply:
If that's the case, we may want to change the logic of this function to:

```js
if ( 'mobile' === context ) { /* extra data for mobile */ } 
if ( 'site-editor' === context ) { /* extra data for site editor */ } 
if ( 'post-editor' === context ) { /* extra data for post editor */ } 

// editor settings for all: mobile, site-editor, post-editor, navigation-editor, etc.
```

Reply:
I think that fix is correct as a defensive measure. One can't guarantee that the key will always be there so let's check for that.

Reply:
> But now I'm wondering if I understood the issue correctly: did you mean that the global styles shouldn't be present in the navigation editor either?

I'm not really sure on an answer for that right now. The navigation editor needs to evolve a bit. Probably in the basic mode the navigation editor doesn't need it, but there's an option for themes to opt in to some more advanced block features, and it might be an option for global styles to work in that case.

This fix works as an interim, so thank you! 👍